compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -171,18 +172,24 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     assert!(!instance.args.has_infer());
 
     let llfn = cx.get_fn(instance);
+    let tcx = cx.tcx();
 
-    let mir = cx.tcx().instance_mir(instance.def);
+    let mir = tcx.instance_mir(instance.def);
+    let mir = instance.instantiate_mir_and_normalize_erasing_regions(
+        tcx,
+        ty::TypingEnv::fully_monomorphized(),
+        ty::EarlyBinder::bind(mir.clone()),
```

Comment:
We need to clone the mir body here, unsure how bad this could be.

Reply:
Maybe in order to avoid affecting performance of things that do not involve `use`, we can just execute this when `tcx.features().ergonomic_clones()` is on.